### PR TITLE
Add GH workflows for release PR w/ checklist & auto merge trunk to develop

### PR DESCRIPTION
**Pull request by tomalec**
See https://github.com/woocommerce/pinterest-for-woocommerce/pull/788 for reference.

### Changes proposed in this Pull Request:

- Add a manual [GH workflow](https://github.com/woocommerce/grow/tree/trunk/packages/github-actions/actions/prepare-extension-release) that takes versions, then starts a branch and creates a PR with the checklist.
The aim is to automate and unify the release process across Grow repos.

- [Add automerge-released-trunk workflow](https://github.com/woocommerce/facebook-for-woocommerce/commit/90e53d9c8779d7abccebd0bee61c31beacd024ed) 
   to automate merging trunk to develop after a release

### Screenshots:

image

### Detailed test instructions:

Unfortunately, you cannot test a manually triggered workflow, until it's merged to the default branch :( 
But this PR follows,
 https://github.com/woocommerce/automatewoo/pull/1435/ and https://github.com/woocommerce/google-listings-and-ads/pull/1982

#### Manual test
1. Create a test repo with the proposed workflow merged to the main branch.
4. Go to `https://github.com/{org}/{repo}/actions/workflows/prepare-release.yml`, like https://github.com/woocommerce/automatewoo/actions/workflows/prepare-release.yml
5. Click the "Run workflow" dropdown, fill in the details, and click "Run..."
6. Wait for the workflow to finish
7. Check the created PR

### Additional details:

1. There is a difference between this flow and the process described at https://github.com/woocommerce/facebook-for-woocommerce/wiki/Build-%26-Release
   Here, we suggest only generating changelog with `woorelease`'s `--generate_changelog` and do not describe [the manual alternative](https://github.com/woocommerce/facebook-for-woocommerce/wiki/Build-%26-Release#preparing-the-release). IMHO, it simplifies the process. Also, I don't know how often you use the "manual-way", so I'm open for feedback.

1. Once the PR is merged, we can update the wiki, to remove the redundant parts, and just state something like we [did in GLA](https://github.com/woocommerce/google-listings-and-ads/wiki/Release-Process#release-basics):

> 1. Go to [Actions > Prepare New Release](https://github.com/woocommerce/google-listings-and-ads/actions/workflows/prepare-release.yml).
> 2. Open the _"Run workflow"_ dropdown, fill in the details, and click _"Run workflow"_ button.
>    image
> 
> 
> 3. Wait for the new release PR to be created by the _github-actions[bot]_.
> 4. Follow the steps from the new PR. Some more hints and troubleshooting are provided below.

### Changelog entry

> Dev - Add release preparation GH workflow.

### CODE REVIEW

**Comment by tomalec** [comment]
Below is the proposed new wiki page. I'll update the image once we have this PR merged/the workflow in place.
Also, I completely removed the section referring to woorelease≤2.0.4, as it's pretty old, and we recommend the latest anyway.


-----


## Release basics

We use GitHub workflow to automate the process. You will still need to use `woorelease` locally and do a few steps manually in GitHub.

1. Go to [Actions > Prepare New Release](https://github.com/woocommerce/facebook-for-woocommerce/actions/workflows/prepare-release.yml).
2. Open the _"Run workflow"_ dropdown, fill in the details, and click _"Run workflow"_ button.
   image



3. Wait for the new release PR to be created by the _github-actions[bot]_.
4. Follow the steps from the new PR. Some more hints and troubleshooting are provided below.

### Prerequisites and Reading

#### GitFlow

We use the GitFlow branching model. Check out the https://venturesp2.wordpress.com/how-we-work/working-with-code/#base-branch if you haven't already.

#### Woorelease

*woorelease* is used to deploy to WooCommerce.com: <https://github.com/woocommerce/woorelease>

**⚠ IMPORTANT** Some Woorelease pre-configuration is necessary:
- Download the `woorelease.zip` from [the latest release](https://href.li/?https://github.com/woocommerce/woorelease/releases/latest), then unzip the archive and install the `woorelease.phar` to use locally.      
- <https://github.com/woocommerce/woorelease/wiki/Prerequisites>
   - Retrieve the Ventures WooCommerce.com API Keys (Secret store → "Team Ventures – WooCommerce.com").
   - Generate your own [GitHub API Keys](https://github.com/settings/tokens).
   - Ensure you have your SSH keys [added to GitHub](https://github.com/settings/keys), as Woorelease uses SSH to clone repos.

#### SVN

Ensure you have `svn` installed on your system by running `which svn`. For macOS, `svn` can be installed (if missing) using `brew install svn`.

SVN requires the following to be present in the `~/.subversion/servers` file:

```ini
[groups]
wporgplugins = plugins.svn.wordpress.org
 
[wporgplugins]
username = woocommerce
store-passwords = no
```

You will likely need to have the `woocommerce` WordPress.org password on hand for your first release.

**Comment by krutidugade** [comment]
I forked this repo as Justin did to test similar [PR here](https://github.com/woocommerce/pinterest-for-woocommerce/pull/788). I created the prepare-release.yml and post-release-automerge.yml files via GH UI. I was expecting permission errors as Justin [mentioned](https://github.com/woocommerce/pinterest-for-woocommerce/pull/788#pullrequestreview-1543199694). Here's a list of errors I got and how I solved them.

```
- Error 1: Process completed with exit code 128.
- Error 2: Unhandled error: HttpError: Resource not accessible by integration
```

I solved this by adding the following block in yml file ([ref](https://sjramblings.io/github-actions-resource-not-accessible-by-integration))

```
permissions:
  id-token: write
  contents: read
  issues: write
  pull-requests: write
```

Later I changed it to `permissions: write-all` after referring to this doc https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs.

`- Error 3: Unhandled error: HttpError: GitHub Actions is not permitted to create or approve pull requests.`

On the Action settings page, I enabled Allow GitHub Actions to create and approve pull requests and Read and Write permissions options under [Workflow permissions](https://github.com/krutidugade/facebook-for-woocommerce/settings/actions/).

Screenshot 2023-07-26 at 10 19 10 PM


`- Error 4: Process completed with exit code 1.`

Screenshot 2023-07-26 at 11 40 29 PM


I get this error and I'm out of ideas on how to proceed. I haven't cloned this repo locally so I'm not sure why it says I may not have changes locally. I tried cloning the repo locally but that didn't help. 

I forked the Pinterest repo and after giving permissions like I did here, it still gives a different error. The Pinterest workflow that works fine on the main repo doesn't work on my forked repo so I'm wondering if I'm missing something.

I have a small suggestion to have Prerequisites first and then release basic content to make the document error-proof.

**Review by krutidugade — COMMENTED** [review]
I have a small suggestion for the Wiki page to have Prerequisites first and then release basic content to make the document error-proof.

**Comment by tomalec** [comment]
Thanks for reviewing and the detailed report :)

I suspect the errors you get is a result of running the workflow to release a version that was already started. Precisely trying to release `1` even though `release/1` branch [already exists](https://github.com/krutidugade/facebook-for-woocommerce/tree/release/1). Unfortunately, the GHAction is not bulletproof for that.

Can you try either releasing a new version, like `1.0.2` or removing the `release/1` branch before running the workflow?

**Comment by krutidugade** [comment]
Thank you, Tomek. You were right. Though I got another error so noting it below for the record.

Error 4 mentioned in the previous comment was resolved by stating a new branch name than existing one.

`- Error 5: Unhandled error: HttpError: Validation Failed: {"resource":"PullRequest","field":"base","code":"invalid"}`

I just had the develop branch. I created the trunk branch which resolved this error.

**Review by krutidugade — APPROVED** [review]
I got the [workflows running](https://github.com/krutidugade/facebook-for-woocommerce/actions). It created [this PR](https://github.com/krutidugade/facebook-for-woocommerce/pull/1) and after merging the PR, it triggered merge from the trunk to develop automatically. Everything works as expected. 
Screenshot 2023-07-27 at 1 01 20 PM

**Pull request merged**
